h/views.py

```diff
@@ -1,16 +1,18 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 import logging
-import re
 
 from pyramid import httpexceptions
 from pyramid.events import ContextFound
 from pyramid.view import forbidden_view_config, notfound_view_config
 from pyramid.view import view_config
+import pyramid.response
 
 from . import session
 from .models import Annotation
 from .resources import Application, Stream
+import h.api_client
```

Comment:
It only makes it necessary to change the patch command if you're importing
the patched symbols themselves, but importing modules should prevent that.

"from .api_client import api_client" should still let you patch anything in
that module.

"from .api_client.api_client import Client" would mean that you have to
patch client locally.
On Mar 20, 2015 8:17 AM, "Nick Stenning" <notifications@github.com> wrote:

> In h/views.py
> <https://github.com/hypothesis/h/pull/2029#discussion_r26835184>:
>
> >
> >  from . import session
> >  from .models import Annotation
> >  from .resources import Application, Stream
> > +import h.api_client
>
> I don't agree that it's a bad idea, I'm afraid, which I why I kept asking
> for it! It doesn't break mock.patch, it just means that you have to patch
> the object where it's used rather than where it is defined, which I've
> never really seen as a problem. If anything, it makes test isolation
> slightly stronger.
>
> It also makes it very clear which modules are local imports, and makes it
> substantially easier to move packages and modules around if you need to.
>
> Lastly, if we do decide we want to adopt a different style, then we should
> do that as group. For the time being, your PR should be consistent with the
> status quo, which in this file is explicit relative imports for local
> modules.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hypothesis/h/pull/2029/files#r26835184>.
>